### PR TITLE
Improve sources definitions suggestions

### PR DESCRIPTION
**Pull request by fivetran-dmitryboykov**

### CODE REVIEW

File: server/src/definition/SourceDefinitionProvider.ts [hunk 1]
```diff
@@ -62,15 +63,30 @@ export class SourceDefinitionProvider implements DbtNodeDefinitionProvider {
...
+          targetRange ?? DbtDefinitionProvider.MAX_RANGE,
+          targetRange ?? DbtDefinitionProvider.MAX_RANGE,
           tableSelectionRange,
         ),
       ];
     }
     return undefined;
   }
+
+  private getSourceRange(lines: string[], tableName: string): Range | undefined {
```
Comment (pgrivachev) [inline server/src/definition/SourceDefinitionProvider.ts:81]: Can we add some unit tests for this method? 

Isn't this process slow when we match each line with regex?

**Review by pgrivachev — APPROVED** [review]
Looks good. We need some tests for regex and ensure that the process to match each line with regex is not slow (probably check on some big file).

File: server/src/definition/SourceDefinitionProvider.ts (hunk 1 above)
Comment (fivetran-dmitryboykov) [reply server/src/definition/SourceDefinitionProvider.ts]: Added unit test.

Comment (fivetran-dmitryboykov) [reply server/src/definition/SourceDefinitionProvider.ts]: Now inclusion of a string will be checked before regex match.

**Review by pgrivachev — APPROVED** [review]
👍

**Pull request merged**